### ISSUE TITLE
Add support for custom "accepts" types

### ISSUE THREAD

**Issue by AndreMaz** [strongloop/loopback#3251]
It seems that there is no way of adding a custom content type for Loopback's API. As stated in [Remoting properties](https://loopback.io/doc/en/lb3/config.json.html#remoting-properties) the supported types are 'application/json','application/javascript', 'application/xml', 'text/javascript', 'text/xml', 'json', 'xml'
The 'application/vnd.api-json' is also supported but is not one of default types.

However, I would like to add a custom content type to my API.

I've tried to add my custom type by changing the  `supportedTypes` in the `server/config.json`.

My `supportedTypes` looks like this.
```javascript
"rest": {
      "supportedTypes": [
            "application/json",
            "application/hsml+json",
      ]
    }
```

However, this does not seem to solve the issue.
Making a request ( `curl -X GET --header 'Accept: application/json' 'http://localhost:3000/api/Endpoints/'`) with `Accept: application/json` works as exptected, i.e, I'm getting the desired result. 

But making a request (`curl -X GET --header 'Accept: application/hsml+json' 'http://localhost:3000/api/Endpoints/`) with `Accept: application/hsml+json` gives me a `406 Not Acceptable`

Is there a way of adding a custom content type to my API?

**TL;DR**: To add a custom JSON-based Content-Type to your API

1. Add your content type to the `supportedTypes` array in the `server/config.json`
```javascript
"rest": {
      "supportedTypes": [
            "application/json",
            "application/customContentType+json",
      ]
    }
```
2. Add a type to the  `json` object in the `server/config.json`
```javascript
    "json": {
      "strict": false,
      "limit": "100kb",
      "type": "application/*+json"
    }
```

3. Add a `case` for your custom Content-Type in the `strong-remoting/lib/http-context.js`'s `HttpContext.prototype.resolveReponseOperation`

```
...
    case 'application/customContentType+json':                         //My Custom Content Type
      result.contentType = 'application/customContentType+json';      //My Custom Content Type
      break;                                                          //My Custom Content Type
...
```

**Comment by AndreMaz** [comment]
Ok I think that I found the source of my problem. It is located in `strong-remoting/lib/http-context.js` https://github.com/strongloop/strong-remoting/blob/5bfaa4742387ac0946873c7bd57b9ea616f43f4c/lib/http-context.js#L315-L350

By default the contentType is set to `application/json` and only changes for `application/vnd.api+json` and `xml`.  

To solve this I've just added my custom content type and it worked without any problems.

```javascript
HttpContext.prototype.resolveReponseOperation = function(accepts) {
  var result = { // default
    sendBody: sendBodyJson,
    contentType: 'application/json',
  };
  switch (accepts) {
    case '*/*':
    case 'application/json':
    case 'json':
      break;
    case 'application/vnd.api+json':
      result.contentType = 'application/vnd.api+json';
      break;
    case 'application/hsml.collection+json':                                  //My Custom Content Type
      result.contentType = 'application/hsml.collection+json';       //My Custom Content Type
      break;                                                                                   //My Custom Content Type
    case 'application/javascript':
    case 'text/javascript':
      result.sendBody = sendBodyJsonp;
      break;
    case 'application/xml':
    case 'text/xml':
    case 'xml':
      if (accepts == 'application/xml') {
        result.contentType = 'application/xml';
      } else {
        result.contentType = 'text/xml';
      }
      result.sendBody = sendBodyXml;
      break;
    default:
      result.sendBody = sendBodyDefault;
      result.contentType = 'text/plain';
      break;
  }
  return result;
};
```

@bajtos @superkhau @davidcheung is there any better way of doing this? Because editing strong-remoting's source code is not very practical

**Comment by bajtos** [comment]
@AndreMaz Makes sense, thank you for investigating the problem in deep.

I agree with you that updating strong-remoting code whenever a new json-based type is needed is not the right solution.

I think we need to come up with some sort of configuration that will allow LoopBack applications to specify custom ContentTypes. For example, what if remote methods could specify the content type they are returning in their metadata? We already allow methods to provide custom status code, see [source](https://github.com/strongloop/strong-remoting/blob/35fd6690b73838fde6925c0f99186afaf5f4102e/lib/http-context.js#L547).

```js
MyModel.remoteMethod('myMethod', {
  accepts: [ /* ... */ ],
  returns: [ /* ... */ ],
  http: {
    path: '/my-method',
    verb: 'get',
    contentType: 'application/hsml+json'
  }
});
```

The thing which is not clear to me is how to handle the case when the client does not specify `application/hsml+json` in the `Accept` request header - should strong-remoting use `application/json` in such case, or reject the request with 406?

Thoughts?

cc @strongloop/sq-lb-epitome @raymondfeng

**Comment by AndreMaz** [comment]
Thanks for answering @bajtos 

I haven't thought about this case because my client always sets the `Accept` header. In my case in particular it would be preferable to receive a `406` because it would indicate that the request is not done properly.

Probably the best solution would be to allow to configure strong-remoting's behavior in case of a missing `Accept` header.
